### PR TITLE
Add converter for `typedef` property 

### LINKED ISSUE #937: Missing Converter for typedef
### 💥 Missing Converter

-   `tslint-to-eslint-config` version: 2.1.0
-   `typescript-eslint` version: 4.14.1

#### TSLint Rule Name

  * tslint-to-eslint-config does not know the ESLint equivalent for TSLint's "typedef".

#### Package Source

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/typedef.md

### PR DESCRIPTION
**Pull request by hyperupcall**
Hoi! ^w^ :wave:

## PR Checklist

-   [x] Fixes #937
-   [x] That issue was marked as [`status: accepting prs`](https://github.com/typescript-eslint/tslint-to-eslint-config/labels/status%3A%20accepting%20prs)

## Overview

The [typedef](https://palantir.github.io/tslint/rules/typedef/) property of TSLint is now considered during the conversion. All config options except `call-signature` and `arrow-call-signature` are accounted for. Options unaccounted for are mentioned in the `notices` array, when applicable

The linked issue and the [ESLint typedef docs](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/typedef.md) mentions that [explicit-function-return-type](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/explicit-function-return-type.md) and [explicit-module-boundary-types](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/explicit-module-boundary-types.md) should be used when converting `call-signature` and `arrow-call-signature`. I did that in a separate branch [here](https://github.com/hyperupcall/tslint-to-eslint-config/commit/b99f54eaa8a4bc1e1fe0729b6b80d5069f77572d), but I wasn't 100% confident that my conversion was correct, so I left it out for now. If it is close to what is considered acceptable, maybe I can add it to this PR?

I'll also add that I had to install `@swc/core` to get things to run, despite [this](65f9090345b31f22371fcdffea22be5b833cbe2dc) mentioning its not required - not sure what's up with that. Also the Docs say Node 12 works, but 14 was the minimum LTS that worked for me (Jest threw `SyntaxError: Unexpected token '.'` when it encountered optional chaining)

### CODE REVIEW

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Awesome, looks good so far. Glad to see this rule finally getting taken care of! 👏

Tests are failing ([ci link](https://app.circleci.com/pipelines/github/typescript-eslint/tslint-to-eslint-config/2825/workflows/a2ef8400-b46d-49c9-a9f0-db9f1da268e6/jobs/14581)) because a lack of coverage on lines `11,15,19,27,31`. You'll want to add >=1 test case that covers them.

**Comment by JoshuaKGoldberg** [comment]
> install @swc/core to get things to run

Huh, sorry about that - I'll send a separate PR to add it in.

**Comment by JoshuaKGoldberg** [comment]
>  Node 12 works, but 14 was the minimum LTS that worked for me (Jest threw SyntaxError: Unexpected token '.' when it encountered optional chaining)

Nice, following up on that too: https://github.com/typescript-eslint/tslint-to-eslint-config/pull/1374

File: src/converters/lintConfigs/rules/ruleConverters/typedef.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    if (tslintRule.ruleArguments.includes("object-destructuring")) {
+        typedefRule.objectDestructuring = true;
+    }
+
+    if (tslintRule.ruleArguments.includes("array-destructuring")) {
+        typedefRule.arrayDestructuring = true;
+    }
+
+    return {
+        ...(tslintRule.ruleArguments.some(
```
[the remark this answers is not in the document]
Comment (hyperupcall) [reply src/converters/lintConfigs/rules/ruleConverters/typedef.ts:39]: Good idea :+1: I'll make those changes

File: src/converters/lintConfigs/rules/ruleConverters/typedef.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            (item) => item === "call-signature" || item === "arrow-call-signature",
+        ) && {
+            notices: ["Options 'call-signature' and 'arrow-call-signature' are ignored"],
```
[the remark this answers is not in the document]
Comment (hyperupcall) [reply src/converters/lintConfigs/rules/ruleConverters/typedef.ts:42]: Cool, I'll add that in too

**Comment by hyperupcall** [comment]
Thank you for your feedback! :smiley:  I will push the improved coverage and address your replies by this Friday (I have some Exams in the meantime)!

**Comment by JoshuaKGoldberg** [comment]
Good luck on the exams! ❤️

**Comment by hyperupcall** [comment]
I've implemented the changes, including use of `Set()`, `Object.entries()`, and got back to 100% codecov :smile: 

Also, both `explicit-function-return-type` and `explicit-module-boundary-types` are now used in the conversion process. Since I'm not super familiar with TSLint, I manually tested each one to be 100% sure conversion was the same. When this conversion is done, I added a message to to `notices` since ESLint seems to only lint all functions (not classifying between arrow functions and non-arrow functions as TSLint does with `arrow-call-signature` and `call-signature`. I tried my best to make it succinct, lemmie know if you have any improvements

**Review by JoshuaKGoldberg — APPROVED** [review]
Super, thanks so much @hyperupcall! 🙌

**Comment by JoshuaKGoldberg** [comment]
This should be available in tslint-to-eslint-config@2.12.0.

**Pull request merged**